refinery/tool_manager/models.py

```diff
@@ -356,13 +341,19 @@ def _launch_workflow(self):
             pointing to the Analysis' status page
         :raises: RuntimeError
         """
+        if self.get_tool_type() != ToolDefinition.WORKFLOW:
+            raise NotImplementedError(
+                "Tool must be of type: {} to launch a workflow".format(
+                    ToolDefinition.WORKFLOW
+                )
```

Comment:
Judgement call, but you could just say "Not a workflow" while hardly inconveniencing the user (and saving 4 lines)... but if you did hit this in the log, you'd probably also want the ID and the tool_type? Something to think about in general, but not worth worrying about here.